src/matlab/homotopy_solver.m

```diff
@@ -39,34 +39,37 @@
     settings_int = varargin{6};
 end
 
-
 import casadi.*
 %%  unfold data
+settings_bkp = settings;
 unfold_struct(settings,'caller')
+settings = settings_bkp;
 unfold_struct(solver_initialization,'caller')
 
 comp_res = model.comp_res;
 nabla_J_fun = model.nabla_J_fun;
 s_elastic_iter = 1;
 
 sigma_k = sigma_0;
-x0 = solver_initialization.lbw(1:model.dimensions.n_x);
+x0 = solver_initialization.w0(1:model.dimensions.n_x);
 
 % lambda00 initialization
 if strcmp(settings.pss_mode, 'Stewart')
-    lambda00 = full(model.lambda00_fun(x0));
+    lambda00 = full(model.lambda00_fun(x0, model.p_global_val));
 elseif strcmp(settings.pss_mode, 'Step')
 %     c_x = full(model.c_fun(x0));
 %     lambda00 = [ max(c_x, 0); min(c_x, 0)];
-    lambda00 = full(model.lambda00_fun(x0));
+    lambda00 = full(model.lambda00_fun(x0, model.p_global_val));
 end
-p_val = [model.p_val(:); lambda00(:)];
+p_val = [model.p_val(:);x0(:); lambda00(:)];
 
 % TODO remove try!
 try
     complementarity_stats = [full(comp_res(w0, p_val))];
 catch
     w0 = w0(1:length(model.w));
+    w0
+    p_val
```

Comment:
These kind of `try`s seem really sketchy.
If you found why its needed maybe add a comment.
But you didnt introduce it of course.

Reply:
I think I might have an explanation. 
At some point not all functions were defined w.r.t. to the correct size of `p_val` as  `p` was modified at several places (e.g., after the introduction of `lambda_00`).
However, comp_res did not depend initially on any `p_val` initially but was defined to depend on it, but with wrong size.

Of course, it should work without a `try` and `catch`. 
Should be an issue anymore, I believe?